src/main/java/duke/util/actions/FetchTasks.java

```diff
@@ -0,0 +1,32 @@
+package duke.util.actions;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import duke.data.exception.DukeException;
+import duke.data.Task;
+import duke.data.exception.DukeExceptionKey;
+import duke.util.TaskList;
+
+public class FetchTasks {
+
+    /**
+     * Fetches all previously created tasks.
+     * @return All tasks in storage.
+     * @throws DukeException When the storage is empty or index is less than 0.
+     */
+    public static List<String> fetchTasks() throws DukeException {
+        List<Task> storage = TaskList.getStorage();
+        List<String> messages = new ArrayList<>();
+        int index = 1;
+        for (Task task : storage) {
```

Comment:
A f[or loop standard ](https://se-education.org/guides/conventions/java/basic.html#loops) where iterator variables are specified could be observed here.